### ISSUE TITLE
Add support for rescheduling and resetting the already scheduled timer.

### ISSUE THREAD

**Issue by gurmitteotia** [gurmitteotia/guflow#22]
As discussed in this [ticket](https://github.com/gurmitteotia/guflow/issues/21) it make sense to have Timer.Reset and Timer.Reschedule API available to library users. These API are quite generic and could be useful in many scenarios. 

Suggested API
```cs
Case 1:

  public class OrderWorkflow : Workflow
    {
        public OrderWorkflow()
        {
            ScheduleTimer("GracePeriod").FireAfter(TimeSpan.FromMinutes(30));
           
            ScheduleActivity<ShipOrder>().AfterTimer("GracePeriod");
        }

       [WorkflowEvent(EventName.Signal)]
        public WorkflowAction Signal(WorkflowSignaledEvent @event)
        {
         //This reschedule timer with by taking the timeout from FireAfter API
          var timer = Timer("GracePeriod");
           if (@event.SignalName == "change_time")
                if (timer.IsActive) timer.Reset(); 
            return Ignore;
        }
    }

Case 2:

  public class OrderWorkflow : Workflow
    {
        public OrderWorkflow()
        {
            ScheduleTimer("GracePeriod").FireAfter(TimeSpan.FromMinutes(30));
           
            ScheduleActivity<ShipOrder>().AfterTimer("GracePeriod");
        }

       [WorkflowEvent(EventName.Signal)]
        public WorkflowAction Signal(WorkflowSignaledEvent @event)
        {
         //This reschedule timer with given timeout.
          var timer = Timer("GracePeriod");
           if (@event.SignalName == "change_time")
                if (timer.IsActive) timer.Reschedule(TimeSpan.FromMinutes(20))); 
            return Ignore;
        }
    }
```

**Comment by alexzm1** [comment]
Any plans on releasing this change in the short term?

**Comment by gurmitteotia** [comment]
I can take this up for next release. Anyway I'm going to focus on timers and signals in next few releases. 
This ticket will also set the stage for next big [ticket](https://github.com/gurmitteotia/guflow/issues/5).

**Comment by gurmitteotia** [comment]
This feature is implemented in version [1.2.15](https://www.nuget.org/packages/Guflow/1.2.15).
